### PR TITLE
[Bug][Connector-v2][ElasticsearchSource] Fix behavior when source empty

### PR DESCRIPTION
**Pull request by CosmosNi**
## Purpose of this pull request

[feature] [Connector-v2][ElasticsearchSource]when source empty,We might want to pull all the fields of Elasticsearch

[Bug][Connector-v2][ElasticsearchSource][When this field instanceof NullNode,The Long field will cast an exception]

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
@iture123 PTAL

**Comment by CosmosNi** [comment]
@iture123 @TyrantLucifer @Hisoka-X PTAL,thanks.

**Review by iture123 — APPROVED** [review]
LGTM

**Review by ic4y — DISMISSED** [review]

**Comment by ic4y** [comment]
@hailin0 @Hisoka-X @EricJoy2048 PTAL

**Comment by Hisoka-X** [comment]
Can you add some test case for this change? Please

**Comment by CosmosNi** [comment]
@Hisoka-X i have already added test case，PTAL

File: seatunnel-connectors-v2/connector-elasticsearch/src/test/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/ElasticsearchSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            SeaTunnelDataType[] fieldTypes = new SeaTunnelDataType[keys.size()];
+            for (int i = 0; i < keys.size(); i++) {
+                String esType = esFieldType.get(keys.get(i));
+                SeaTunnelDataType seaTunnelDataType =
+                        elasticSearchDataTypeConvertor.toSeaTunnelType(esType);
+                fieldTypes[i] = seaTunnelDataType;
+            }
+            rowTypeInfo = new SeaTunnelRowType(keys.toArray(new String[0]), fieldTypes);
+        }
+        Assertions.assertNotNull(rowTypeInfo);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/test/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/ElasticsearchSourceTest.java:61]: Please assert it with right `rowTypeInfo`, not assertNotNull.

Comment (CosmosNi) [reply seatunnel-connectors-v2/connector-elasticsearch/src/test/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/ElasticsearchSourceTest.java:61]: It has been corrected

**Comment by Carl-Zhou-CN** [comment]
@CosmosNi  hi, is there any new progress on it

**Comment by CosmosNi** [comment]
Due to submission email error，close merge

**Pull request closed without merging**